haystack/components/tools/tool_invoker.py

```diff
@@ -184,15 +184,17 @@ def __init__(
 
         # Convert Toolset to list for internal use
         if isinstance(tools, Toolset):
-            tools = list(tools)
+            converted_tools = list(tools)
+        else:
+            converted_tools = tools
```

Comment:
I made a slight change here so we don't overwrite/change the `tools` input from the user.